app/controllers/users_controller.rb

```diff
@@ -1,12 +1,10 @@
 class UsersController < ApplicationController
-  def show
-    @user = User.find_by params[:id]
+  before_action :logged_in_user, except: %i(new create show)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: :destroy
+  before_action :user_exist, except: %i(index new create)
```

Comment:
```suggestion
  before_action :user_exist, except: %i(index new create)
  before_action :correct_user, only: %i(edit update)
  before_action :admin_user, only: :destroy
```
call back cũng có thứ tự chạy đấy cụ nhé